### PR TITLE
core(legacy): convert some base artifacts to regular gatherers

### PR DESCRIPTION
**Pull request by adamraine**
Fixes https://github.com/GoogleChrome/lighthouse/issues/14678

I thought about putting bf-cache-failures in it's own pass, but that pass would probably require us to load the entire page without any shortcuts and/or require heavy modification to the legacy gather runner.

Instead, I chose to make the legacy runner more like the FR runner and just convert some base artifacts into regular artifacts.

### CODE REVIEW

File: core/legacy/gather/gather-runner.js [hunk 1]
```diff
@@ -424,37 +418,6 @@ class GatherRunner {
...
-    } catch (err) {
-      log.error('GatherRunner WebAppManifest', err);
-      baseArtifacts.WebAppManifest = null;
-    }
-
-    try {
-      baseArtifacts.InstallabilityErrors = await InstallabilityErrors.getInstallabilityErrors(
-        passContext.driver.defaultSession);
-    } catch (err) {
-      log.error('GatherRunner InstallabilityErrors', err);
```
Comment (connorjclark) [inline core/legacy/gather/gather-runner.js:440]: relevant context #13312

I don't know if this is still needed but ... can we replicate this in the gatherer?

File: core/legacy/gather/gather-runner.js [hunk 2]
```diff
@@ -424,37 +418,6 @@ class GatherRunner {
...
-          },
-        ],
-      };
-    }
-
-    try {
-      baseArtifacts.Stacks = await Stacks.collectStacks(passContext.driver.executionContext);
-    } catch (err) {
-      log.error('GatherRunner Stacks', err);
-      baseArtifacts.Stacks = [];
```
Comment (connorjclark) [inline core/legacy/gather/gather-runner.js:455]: I'm not sure what happens now if this artifact errors. Probably similar to above, we need to try/catch and set a default in the gatherer such that `artifacts.Stacks` will never be undefined (or an Error).

File: core/lib/stack-packs.js [hunk 3]
```diff
@@ -63,10 +63,12 @@ const stackPacksToInclude = [
 
 /**
  * Returns all packs that match the stacks found in the page.
- * @param {LH.Artifacts['Stacks']} pageStacks
+ * @param {LH.Artifacts['Stacks']|undefined} pageStacks
  * @return {LH.RawIcu<Array<LH.Result.StackPack>>}
  */
 function getStackPacks(pageStacks) {
+  if (!pageStacks) return [];
```
Comment (adamraine) [inline core/lib/stack-packs.js:70]: We can reach this condition on error reports now

**Review by connorjclark — APPROVED** [review]
Yay for less flaky smokes!

File: core/lib/stack-packs.js (hunk 3 above)
Comment (connorjclark) [reply core/lib/stack-packs.js:70]: Isn't that just because of removing the initial values in initializeBaseArtifacts ? Maybe add them back?

Comment (adamraine) [reply core/lib/stack-packs.js:70]: We shouldn't use `initializeBaseArtifacts` to subtly create a default value for some artifacts but not others IMO

**Pull request merged**